### PR TITLE
[RFE} Test for upgrading satellite to 6.16 SCA only

### PR DESCRIPTION
**Pull request by ColeHiggins2**
Automation for SCA Only

### CODE REVIEW

**Review by lhellebr — COMMENTED** [review]
This is a draft, yet it's marked for review. I've skimmed through it quickly and don't see any obvious issues.

File: tests/upgrades/test_repository.py [hunk 1]
```diff
@@ -371,3 +372,151 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
...
+    def test_pre_simple_content_access_only(
+        self,
+        target_sat,
+        save_test_data,
+        rhel_contenthost,
+        upgrade_entitlement_manifest_org,
+        upgrade_entitlement_manifest,
+    ):
+        """Register host with an activation key that has one custom repository(overriden to enabled)
+        and one red rat repository enabled but no subscriptions attached. Assert that only
```
Comment (vsedmik) [inline tests/upgrades/test_repository.py:395]: lol
```suggestion
        and one Red Hat repository enabled but no subscriptions attached. Assert that only
```

File: tests/upgrades/test_repository.py [hunk 2]
```diff
@@ -371,3 +372,151 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
...
+        for content in all_content:
+            if content['name'] == custom_repo.name:
+                custom_content_label = content['label']
+            if content['repositories'][0]['id'] == rh_repo_id:
+                rh_content_label = content['label']
+        activation_key.content_override(
+            data={'content_overrides': [{'content_label': custom_content_label, 'value': '1'}]}
+        )
+        rhel_contenthost.install_katello_ca(target_sat)
+        rhel_contenthost.register_contenthost(org=org.name, activation_key=activation_key.name)
```
Comment (vsedmik) [inline tests/upgrades/test_repository.py:458]:
```suggestion
        rhel_contenthost.register(org=org, loc=None, activation_key=activation_key.name, target=target_sat)
```
Let's go with global registration, it should work the same.

File: tests/upgrades/test_repository.py [hunk 3]
```diff
@@ -371,3 +372,151 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
...
+        assert custom_content_label in enabled.stdout
+        sca_access = target_sat.execute(
+            f'echo "Organization.find({org_id}).simple_content_access?" | foreman-rake console'
+        )
+        assert 'True' in sca_access.stdout
+        sca_mode = target_sat.execute(
+            f'echo "Organization.find({org_id}).content_access_mode" | foreman-rake console'
+        )
+        assert 'entitlement' not in sca_mode.stdout
+        assert 'org_environment' in sca_mode.stdout
```
Comment (vsedmik) [inline tests/upgrades/test_repository.py:517]: Does this stand for "SCA-only"? I wonder why it's called "org_environment".

**Review by vsedmik — CHANGES_REQUESTED** [review]
Looks pretty good, only one small suggestion to use GR.

**Review by vsedmik — APPROVED** [review]
Rebase needed

**Review by damoore044 — APPROVED** [review]
ack looks good after rebase

File: tests/upgrades/test_repository.py [hunk 4]
```diff
@@ -371,3 +372,152 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
...
+                custom_content_label = content['label']
+            if content['repositories'][0]['id'] == rh_repo_id:
+                rh_content_label = content['label']
+        activation_key.content_override(
+            data={'content_overrides': [{'content_label': custom_content_label, 'value': '1'}]}
+        )
+        rhel_contenthost.register(
+            org=org, loc=None, activation_key=activation_key.name, target=target_sat
+        )
+        enabled = rhel_contenthost.execute("subscription-manager repos --list-enabled")
```
Comment (vijaysawant) [inline tests/upgrades/test_repository.py:460]: making string format consistant
```suggestion
        enabled = rhel_contenthost.execute('subscription-manager repos --list-enabled')
```

File: tests/upgrades/test_repository.py [hunk 5]
```diff
@@ -371,3 +372,152 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
...
+        :expectedresults:
+            1. Content host is now in simple content access mode and custom repository
+            is enabled on host.
+        """
+        client_hostname = pre_upgrade_data.get('rhel_client')
+        org_id = pre_upgrade_data.get('org_id')
+        rh_content_label = pre_upgrade_data.get('rh_content_label')
+        custom_content_label = pre_upgrade_data.get('custom_content_label')
+        rhel_client = ContentHost.get_host_by_hostname(client_hostname)
+        enabled = rhel_client.execute("subscription-manager repos --list-enabled")
```
Comment (vijaysawant) [inline tests/upgrades/test_repository.py:507]: making string format consistant
```suggestion
        enabled = rhel_client.execute('subscription-manager repos --list-enabled')
```

**Review by vijaysawant — CHANGES_REQUESTED** [review]
String format

File: tests/upgrades/test_repository.py [hunk 6]
```diff
@@ -371,3 +372,152 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
...
+        self,
+        target_sat,
+        save_test_data,
+        rhel_contenthost,
+        upgrade_entitlement_manifest_org,
+        upgrade_entitlement_manifest,
+    ):
+        """Register host with an activation key that has one custom repository(overriden to enabled)
+        and one Red Hat repository enabled but no subscriptions attached. Assert that only
+        the red hat repository is enabled on the host and that the host is in entitlement mode.
```
Comment (vijaysawant) [inline tests/upgrades/test_repository.py:396]:
```suggestion
        the Red Hat repository is enabled on the host and that the host is in entitlement mode.
```

File: tests/upgrades/test_repository.py [hunk 7]
```diff
@@ -371,3 +372,152 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
...
+
+        :id: preupgrade-d6661342-a348-4dd5-9ddf-3fd630b1e286
+
+        :steps:
+            1. Before satellite upgrade
+            2. Create new organization and location
+            3. Upload a manifest in it
+            4. Create a ak with repos and no subscriptions added in it
```
Comment (vijaysawant) [inline tests/upgrades/test_repository.py:404]: which one is correct `added`/`attached`? non-blocking comment
```suggestion
            4. Create a ak with repos and no subscriptions attached in it
```

Comment (ColeHiggins2) [reply tests/upgrades/test_repository.py]: you add subscriptions not attach, but ill fix the grammar

Comment (vijaysawant) [reply tests/upgrades/test_repository.py]: okay

File: tests/upgrades/test_repository.py [hunk 8]
```diff
@@ -372,3 +373,152 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
...
+            6. Register the content host
+            7. Confirm content host only has red hat repo enabled
+            8. Confirm content host is in entitlement mode
+
+        :expectedresults:
+            1. Content host is registered in entitlement mode and has only the red hat repository
+            enabled on host.
+
+        """
+        _, manifester = upgrade_entitlement_manifest
```
Comment (Griffin-Sullivan) [inline tests/upgrades/test_repository.py:416]: Doesn't look like you're using this variable anywhere

File: tests/upgrades/test_repository.py [hunk 9]
```diff
@@ -372,3 +373,152 @@ def test_post_scenario_sync_large_repo(self, target_sat, pre_upgrade_data):
...
+        ).create()
+        custom_repo.sync()
+        if rhel_contenthost.os_version.major > 7:
+            rh_repo_id = target_sat.api_factory.enable_sync_redhat_repo(
+                constants.REPOS[f'rhel{rhel_contenthost.os_version.major}_bos'], org.id
+            )
+        else:
+            rh_repo_id = target_sat.api_factory.enable_sync_redhat_repo(
+                constants.REPOS[f'rhel{rhel_contenthost.os_version.major}'], org.id
+            )
```
Comment (Griffin-Sullivan) [inline tests/upgrades/test_repository.py:432]: We really ought to have a method for doing this in `enable_sync_redhat_repo`.

**Review by Griffin-Sullivan — APPROVED** [review]
Looks good to me. Non-blocking comments. Did you test this locally? I assume PRT can't test this.

**Comment by ColeHiggins2** [comment]
============================= test session starts ==============================
platform linux -- Python 3.11.2, pytest-8.1.1, pluggy-1.4.0
Mandatory Requirements Mismatch: jinja2==3.1.4 pytest-order==1.2.1 PyYAML==6.0.1 cryptography==42.0.7 nailgun @ git+https://github.com/SatelliteQE/nailgun.git@master#egg=nailgun pytest-ibutsu==2.2.4 navmazing==1.2.2 pytest-fixturecollection==0.1.2 python-box==7.1.1 wrapanapi==3.6.0 pytest-xdist==3.6.1 pytest-mock==3.14.0 pytest-services==2.2.1 betelgeuse==1.11.0 apypie==0.4.0 pytest-reportportal==5.4.1 fauxfactory==3.1.1 pyotp==2.9.0 wait-for==1.2.0 dynaconf[vault]==3.2.5 requests==2.32.3 broker[docker]==0.4.9 airgun @ git+https://github.com/SatelliteQE/airgun.git@master#egg=airgun pytest==8.2.1 testimony==2.4.0 deepdiff==7.0.1 tenacity==8.3.0 productmd==1.38 manifester==0.2.3
Optional Requirements Mismatch: pre-commit==3.7.1 sphinx==7.3.7 manage==0.1.15 pytest-cov==5.0.0 redis==5.0.4 flake8==7.0.0 sphinx-autoapi==3.1.1
To update mismatched requirements, run the pytest command with '--update-required-reqs' OR '--update-all-reqs' option.
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/colehiggins/projects/robottelo
configfile: pyproject.toml
plugins: services-2.2.1, ibutsu-2.2.4, anyio-4.2.0, xdist-3.5.0, order-1.2.0, mock-3.14.0, fixturecollection-0.1.2, cov-5.0.0, reportportal-5.4.1
collected 3 items
.
.
.
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================= 3 passed, 356 warnings in 2223.20s (0:37:03) =================
Finished running tests!

**Pull request merged**